linker.go

```diff
@@ -136,5 +136,22 @@ func fixupJumpsAndCalls(insns asm.Instructions) error {
 		}
 	}
 
+	// fixupBPFCalls replaces bpf_probe_read_{kernel,user}[_str] with bpf_probe_read[_str] on older kernels
+	// https://github.com/libbpf/libbpf/blob/master/src/libbpf.c#L6009
+	iter = insns.Iterate()
+	for iter.Next() {
+		ins := iter.Ins
+		switch asm.BuiltinFunc(ins.Constant) {
```

Comment:
This is only valid to do if `ins.IsFunctionCall` is true.

Reply:
You're right, I will add an if-filter here.
However, `ins.IsFunctionCall()` returns false for the FnProbeReadKernel, because ins.Src equals to R0. Can I use a `ins.OpCode.JumpOp() == Call` instead?
https://github.com/cilium/ebpf/blob/master/asm/instruction.go#L177-L182

Reply:
Updated.

Reply:
Good point! Can you add `IsBuiltinCall` which asserts `JumpOp == Call` and `ins.Src == R0` and use that?

Reply:
Nice suggestion!
I just refer to libbpf and found they have additional ins.Dst == R0 check, so I added this as well.